From: Lucas Stach <l.stach@pengutronix.de>
Date: Mon, 5 Oct 2020 16:45:11 +0200
Subject: [PATCH] build: fix the debug parameter to properly handle
 --disable-debug

Right now the the enableval isn't checked, so the debug options are added
for --enable-debug and --disable-debug, which is quite surprising for a
user. Fix this to peroply handle the disable case.
---
 configure.ac | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 6d9b04b45cb2..f1bb7b14eae4 100644
--- a/configure.ac
+++ b/configure.ac
@@ -42,8 +42,11 @@ m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
 AC_ARG_ENABLE(debug,
               AS_HELP_STRING([--enable-debug],
                 [Enable debugging information]),
-              [CFLAGS="$CFLAGS -g -O0 -DDEBUG"
-               CXXFLAGS="$CXXFLAGS -g -O0 -DDEBUG"],
+              [if test x$enableval = xyes; then
+               CFLAGS="$CFLAGS -g -O0 -DDEBUG"
+               CXXFLAGS="$CXXFLAGS -g -O0 -DDEBUG"; else
+               CFLAGS="$CFLAGS -DNDEBUG"
+               CXXFLAGS="$CXXFLAGS -DNDEBUG"; fi],
               [CFLAGS="$CFLAGS -DNDEBUG"
                CXXFLAGS="$CXXFLAGS -DNDEBUG"])
 
